### ISSUE TITLE
Move Portal "UserChangeLog" and "UserUpdate" from the portal to the API / Core modules

### ISSUE THREAD

**Issue by pauljamescleary** [vinyldns/vinyldns#116]
**Motivation**
Inserting new users as well as auditing user changes is currently done in the portal.  This was done as the portal was originally a separate repository and there was no code sharing done with the api.

Now that the api and portal are in the same repo, and there is code sharing via the `core` module, we should move the repositories out of the portal and into an official "Repository" in the `core` module.

**Scope of change**

1. Remove change logs and user insert out of the portal.  This would remove all of the `datastores`
1. Move the `UserRepository` to the `core` module
1. Add missing functions to the `UserRepository` that are needed by the portal
1. Add the `UserChangeRepository` to the `core` module
1. Update the configuration for the portal to use the same configuration as the API

**Detail**
There is quite a bit of detail for this.  Will try to be explicit and answer questions along the way.

**Step 1**
Move the `User` model to the `core` module under `domain.membership`.

Create a `UserChange` type next to `User` that has the same attributes as the `UserChangeMessage` from the portal.

**Step 2**
Create a `UserChangeRepository` in the `core` module.  This currently lives in the portal under `app/controllers/datastores/DynamoDBChangeLogStore`.    It should live in core under `domain.user` namespace.

The important function is called `log` here, it takes a `ChangeMessage` which is not consistent with the api repositories.  Also, there is only a single `ChangeMessage` type called `UserChangeMessage`, so we can remove the `ChangeMessage` trait all together.  

The following is the result of the move:

```scala
trait UserChangeRepository {
  def save(userChange: UserChange): Future[UserChange]
}
```

**Step 3**
Create a `DynamoDBUserChangeRepository` that implements our interface.  It should live in core under `repository.dynamodb`

To provide this implementation, you will need to move `DynamoDBClient` and `DynamoDBHelper` into the core module as well.  They should live in `repository.dynamodb`

**Step 4**
Have the portal use the new `DynamoDBUserChangeRepository` instead of the old `DynamoDBChangeLogStore`.  

This will require a change to how the store is loaded in the `Module`, as well as how it is injected.  These are tedious changes but straightforward.

**Step 5**
Move the `UserRepository` to the `core` module.  This would include the trait, types, and DynamoDBUserRepository. 

**Step 6**
Update the portal to use the `DynamoDBUserRepository` from core, replacing the `UserAccountStore` that it uses now

**Step 7**
Delete the things that are no longer used from the portal.  We do not need the `InMemory` stores there anymore since everything runs in an assumed docker environment.

**Comment by britneywright** [comment]
Do you imagine each step being a separate issue?

**Comment by pauljamescleary** [comment]
I guess it depends on how we size it.  I think if the whole thing is under an 8, just keep it as a single issue and then split it up by PRs.

Certainly, there are 2 distinct moves, so they can be done in parallel.   The 2 moves are ChangeLog --> Core, and the other is replacing the `UserAccountStore` with `DynamoDBUserRepository`.

**Comment by britneywright** [comment]
I like that split.

**Comment by rebstar6** [comment]
can be a separate card, but removing the in memory repos from the portal is a related task to this work

**Comment by pauljamescleary** [comment]
@rebstar6 correct, I will make sure that is clear.  Is everyone ok with moving this forward?

**Comment by pauljamescleary** [comment]
I'm going to +1 this, as I think the detail is sufficient to size.

**Comment by britneywright** [comment]
+1

**Comment by pauljamescleary** [comment]
Moving this to blocked for right now.  Not sure which specific issue blocks this.

We need to port most of our domain (at least case classes / models and repositories) into core, along with things like `DynamoDBHelper`.

That is an extremely simple refactor, but it must take place before this card can go any further.
